.github/workflows/ansible-lint.yml

```diff
@@ -0,0 +1,30 @@
+name: Ansible Lint
+
+on: [push, pull_request]
+
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+      with:
+        lfs: false # not needed currently
+    - name: Lint Ansible Playbook
+      uses: ansible/ansible-lint-action@master
+      with:
+        targets: |
+          tools/scripts/ansible/setup-playbook.yml
+        override-deps: |
+          ansible==2.9
+          ansible-lint==4.2.0
+        args: ""
+  build:
```

Comment:
Does the build job depend on the lint job succeeding, or are they independent?  Asking b/c I don't know Ansible, so I can't say if linting is more like validation / error-prone checking or if it is more like formatting style.

If it linting is more than just strongly recommended (ex: in the ICU4X case, we treat warnings like errors), then you could make the build job depend on the lint job.

But if the the build job or lint job take a long time, and/or one job's input does not necessarily depend on the success of output of the other job, then it makes sense to keep them un-linked as deps.

Reply:
> Does the build job depend on the lint job succeeding, or are they independent? Asking b/c I don't know Ansible, so I can't say if linting is more like validation / error-prone checking or if it is more like formatting style.

They are independent.   Amusingly, only the _mac_ Github VMs have VT-x enabled to sublaunch an Ubuntu VM with vagrant. (Go figure, that's what Github support said somewhere.)  But the ansible-lint action is only Linux.

The linting is validation/error prone and deprecation checking. It “made” me rewrite a 'notifier' to become a 'handler'.  The build step _could_ pass if lint fails, but at this point I'd like to keep the linter happy since i'm newer to ansible. (I have been using it personally for a couple of years, but not as part of CD.)

>  If it linting is more than just strongly recommended (ex: in the ICU4X case, we treat warnings like errors), then you could make the build job depend on the lint job.

Does that mean running them in series?

> But if the the build job or lint job take a long time, and/or one job's input does not necessarily depend on the success of output of the other job, then it makes sense to keep them un-linked as deps.

The build job takes a while because it spins up a VM.  Actually, one way I will want to refactor this, and I'm not sure entirely how, is to make the 'build' step deploy the `cldr-apps.war` artifact from the 'ant' workflow. It seems you [cannot share artifacts across workflows](https://stackoverflow.com/questions/60355925/share-artifacts-between-workflows-github-actions), so this 'build' job may want to move to be a later step in the 'ant' workflow (which you reviewed the other day).  However, it would be good to make the 'ant' workflow more speedy before that, perhaps by making all of the long running tests a parallel step in 'ant'.

I'll do this. I'll draw out a simple flowchart of where I think the build steps and artifacts need to go. One of the tensions here is premature optimization vs. caching.  I wince at all of those GB of data, downloaded anew each time, but I think the reason is because the buildbots are of course stateless independent throwaway units.  So this is kind of the 'new normal' for distributed cloud based building.

Reply:
@echeran I'm inclined to try to merge this as is if there aren't issues, because it does bring an improvement, and then to iterate.

Reply:
Sounds good to me.